raiden/network/transport/matrix/client.py

```diff
@@ -432,38 +447,41 @@ def get_user_presence(self, user_id: str) -> str:
     def call(callback, *args, **kwargs):
         return callback(*args, **kwargs)
 
-    def _sync(self, timeout_ms=30000):
+    def _sync(self, timeout_ms=SYNC_TIMEOUT_MS):
         """ Reimplements MatrixClient._sync, add 'account_data' support to /sync """
         log.debug(
             "Sync called", node=node_address_from_userid(self.user_id), current_user=self.user_id
         )
 
         response = self.api.sync(self.sync_token, timeout_ms)
-        prev_sync_token = self.sync_token
-        self.sync_token = response["next_batch"]
-
-        if self._handle_thread is not None:
-            # if previous _handle_thread is still running, wait for it and re-raise if needed
-            self._handle_thread.get()
 
-        is_first_sync = prev_sync_token is None
-        self._handle_thread = gevent.Greenlet(self._handle_response, response, is_first_sync)
-        self._handle_thread.name = (
-            f"GMatrixClient._sync user_id:{self.user_id} sync_token:{prev_sync_token}"
-        )
-        self._handle_thread.link_exception(lambda g: self.sync_thread.kill(g.exception))
-        log.debug(
-            "Starting handle greenlet",
-            node=node_address_from_userid(self.user_id),
-            first_sync=is_first_sync,
-            sync_token=prev_sync_token,
-            current_user=self.user_id,
-        )
-        self._handle_thread.start()
+        self.response_queue.put(response)
+        self.sync_token = response["next_batch"]
 
         if self._post_hook_func is not None:
             self._post_hook_func(self.sync_token)
 
+    def _handle_process_worker(self, response_queue: NotifyingQueue, event_stop: Event) -> None:
```

Comment:
Can you add a small docstring here explaining in one line the function's purpose?